feat(curriculum): adding graphs and trees quiz questions (#62264)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/quiz-graphs-and-trees/67f4138c34cd9c4ac783ceef.md

An algorithm that visits all nodes in one direction before moving another direction in the graph.

---

An algorithm that visits four nodes at a time before moving to the next level in the graph.

#### --answer--

An algorithm that visits all neighboring nodes before moving to the next level in the graph.

### --question--

#### --text--

Which of the following algorithms will follow each branch as deep as possible before it backtracks?

#### --distractors--

Data-First Search

---

Dock-First Search

---

Daemon-First Search

#### --answer--

Depth-First Search

### --question--

#### --text--

Which of the following terms represents a two-dimensional list in which the rows and columns represent the graph's vertices?

#### --distractors--

Array matrix

---

Depth matrix

---

Vertex matrix

#### --answer--

Adjacency matrix

### --question--

#### --text--

Which of the following terms represents an array or dictionary that stores all the neighbors of each node?

#### --distractors--

Cyclic list

---

Tuple list

---

Breadth list

#### --answer--

Adjacency list

### --question--

#### --text--

What must be true for a graph to be classified as a tree?

#### --distractors--

> A tree must have at least one loop or cycle.

Reviewer: For the way the question is structured I think it's better
```suggestion
It must have at least one loop or cycle.
```

---

> A tree must have at least two loops or cycles.

Reviewer: ```suggestion
It must have at least two loops or cycles.
```

---

> A tree must have three loops or cycles.

Reviewer: ```suggestion
It must have three loops or cycles.
```

#### --answer--

> A tree cannot have loops or cycles.

Reviewer: ```suggestion
It cannot have loops or cycles.
```
